CLAUDE.md

```diff
@@ -0,0 +1,200 @@
+# CLAUDE.md - Ethereum Execution Spec Tests
+
+> **CRITICAL**: This repository aims to provide excellent tooling for generating JSON test vectors that test Ethereum execution layer clients. Correctness is absolute priority. The repo prioritizes a contributor-first mindset.
+
+## 🎯 Core Purpose
+
+- `./tests/`: Python tests that **generate JSON test vectors (fixtures)** via `fill` command
+- `./src/pytest_plugins/filler/`: Implements `fill` command (test vector generation from Python source)
+- `./src/pytest_plugins/consume/`: Implements `consume` command (test vector execution)
+- `./src/pytest_plugins/execute/`: Implements `execute` command (live JSON-RPC testing from Python source)
+- `./src/ethereum_test_*`: Core framework libraries and data structures
+
+### Key Terminology (CRITICAL)
+
+**"Fixtures" has TWO meanings:**
+
+1. **Test Fixtures** (JSON files) - The test vectors this framework generates
+2. **Pytest Fixtures** - Standard pytest setup/teardown (`pre`, `state_test`, etc.)
+
+### Workflows
+
+```text
+Fill/Consume: Python Tests → fill → JSON Fixtures → consume → Client Testing
+Execute: Python Tests → execute → Live JSON-RPC Testing
+```
+
+## 🚀 Essential Commands
+
+All commands use `uv run` prefix.
+
+### Setup
+
+```bash
+uv sync --all-extras
+uv run solc-select use 0.8.24 --always-install
+uvx pre-commit install
+```
+
+### Core Workflow
+
+```bash
+# Create test
+uv run eest make test
+
+# Generate fixtures (PRIMARY WORKFLOW)
+uv run fill --fork=Cancun path/to/test.py --clean -v
```

Comment:
```suggestion
uv run fill --fork=Prague path/to/test.py --clean -v -m "not slow"
```

Ideally we could provide another command to get the latest development fork, but I think this should be ok for first-time contributors.